editor/src/components/canvas/canvas-strategies/keyboard-absolute-move-resize.spec.browser2.tsx

```diff
@@ -0,0 +1,266 @@
+/* eslint-disable jest/expect-expect */
+import { act, RenderResult } from '@testing-library/react'
+import sinon, { SinonFakeTimers } from 'sinon'
+
+import * as EP from '../../../core/shared/element-path'
+import { isFeatureEnabled, setFeatureEnabled } from '../../../utils/feature-switches'
+import { wait } from '../../../utils/utils.test-utils'
+import { selectComponents } from '../../editor/actions/action-creators'
+import { getPrintedUiJsCode, renderTestEditorWithCode } from '../ui-jsx.test-utils'
+import { KeyboardInteractionTimeout } from './interaction-state'
+
+describe('Keyboard Absolute Strategies E2E', () => {
+  let originalCanvasStrategiesFSValue: boolean
+  before(() => {
+    viewport.set(2200, 1000)
+    originalCanvasStrategiesFSValue = isFeatureEnabled('Canvas Strategies')
+    setFeatureEnabled('Canvas Strategies', true)
+  })
+  after(() => {
+    setFeatureEnabled('Canvas Strategies', originalCanvasStrategiesFSValue)
+  })
+
+  let clock: SinonFakeTimers
+  beforeEach(function () {
+    setFeatureEnabled('Canvas Strategies', true)
+    clock = sinon.useFakeTimers({
+      // the timers will tick so the editor is not totally broken, but we can fast-forward time at will
+      // WARNING: the Sinon fake timers will advance in 20ms increments
+      shouldAdvanceTime: true,
+    })
+  })
+  afterEach(function () {
+    clock.restore()
+  })
+
+  it('Pressing Shift + ArrowRight 3 times', async () => {
+    const { expectElementLeftOnScreen, expectElementLeftInPrintedCode } = await setupTest()
+
+    pressArrowRightHoldingShift3x()
+    expectElementLeftOnScreen(30)
+
+    // tick the clock so useClearKeyboardInteraction is fired
+    clock.tick(KeyboardInteractionTimeout)
+    await expectElementLeftInPrintedCode(30)
+  })
+
+  it('Pressing Shift + ArrowRight 3 times, then pressing Esc before the keyboard strategy timer succeeds will cancel the strategy', async () => {
+    const { expectElementLeftOnScreen, expectElementLeftInPrintedCode } = await setupTest()
+
+    pressArrowRightHoldingShift3x()
+    // the element visually moved 30 pixels to the right on screen
+    expectElementLeftOnScreen(30)
+
+    // tick the clock so useClearKeyboardInteraction is fired
+    clock.tick(KeyboardInteractionTimeout)
```

Comment:
this will move the clock forward by "3000ms", to make sure the setTimeout in useClearKeyboardInteraction is called. it takes no real-time to do so, making the test pretty fast!